central/complianceoperator/v2/report/manager/generator/report_gen_impl_test.go

```diff
@@ -105,7 +110,37 @@ func (s *ComplainceReportingTestSuite) TestProcessReportRequest() {
 		s.Require().Error(s.reportGen.ProcessReportRequest(request))
 	})
 
-	s.Run("Fail to upsert Snapshot", func() {
+	s.Run("FormatCSVReport error", func() {
+		s.snapshotDS.EXPECT().GetSnapshot(gomock.Any(), gomock.Any()).Times(1).
+			Return(&storage.ComplianceOperatorReportSnapshotV2{
+				ReportStatus: &storage.ComplianceOperatorReportStatus{},
+			}, true, nil)
+		s.resultsAggregator.EXPECT().GetReportData(gomock.Any()).Times(1).Return(&report.Results{})
+		s.formatter.EXPECT().FormatCSVReport(gomock.Any(), gomock.Any()).Times(1).Return(nil, errors.New("some error"))
+		s.snapshotDS.EXPECT().UpsertSnapshot(gomock.Any(), gomock.Any()).Times(1).
+			DoAndReturn(func(_ any, snapshot *storage.ComplianceOperatorReportSnapshotV2) error {
+				s.Require().Equal(storage.ComplianceOperatorReportStatus_FAILURE, snapshot.GetReportStatus().GetRunState())
+				return nil
+			})
```

Comment:
Removing the entire `DoAndRetrun` part would also make this test pass. Is it here only to make the assertion?

Reply:
Yeah it was only for the assertion. I've changed all the cases to use `gomock.Cond` instead (when possible).